platform/mellanox/mlnx-platform-api/sonic_platform/modules_mgmt.py

```diff
@@ -458,6 +471,32 @@ def power_on_module(self, port, module_sm_obj, dynamic=False):
                 return STATE_HW_NOT_PRESENT
         return STATE_NOT_POWERED
 
+    def is_supported_for_software_control(self, xcvr_api):
+        return isinstance(xcvr_api, cmis.CmisApi) or isinstance(xcvr_api, sff8636.Sff8636Api) or isinstance(xcvr_api, sff8436.Sff8436Api)
+
+    def update_frequency(self, port, xcvr_api):
+        # first read the frequency support - if it's 1 then continue, if it's 0 no need to do anything
+        module_fd_freq_support_path = SYSFS_INDEPENDENT_FD_FREQ_SUPPORT.format(port)
+        val_int = utils.read_int_from_file(module_fd_freq_support_path)
+        if 1 == val_int:
+            if isinstance(xcvr_api, cmis.CmisApi):
+                # for CMIS modules, read the module maximum supported clock of Management Comm Interface (MCI) from module EEPROM.
+                # from byte 2 bits 3-2:
+                # 00b means module supports up to 400KHz
+                # 01b means module supports up to 1MHz
+                logger.log_debug(f"check_module_type reading mci max frequency for port {port}")
+                read_mci = xcvr_api.xcvr_eeprom.read_raw(CMIS_MCI_EEPROM_OFFSET, 1)
+                logger.log_debug(f"check_module_type read mci max frequency {read_mci} for port {port}")
+                frequency_bits = read_mci & CMIS_MCI_MASK
+            elif isinstance(xcvr_api, sff8636.Sff8636Api) or isinstance(xcvr_api, sff8436.Sff8436Api):
+                # for SFF modules, frequency is always 400KHz
+                frequency_bits = 0b00
```

Comment:
should it be a value like 0 or 1? Please see comment at line#496

Reply:
It indeed should be 0 or 1, but according to Andy, for SFF modules there is no special advertisement for for the frequency in the EEPROM (like there is for CMIS modules). He says that its always 400KHz for SFF modules, so according to line #496 the value should be '00'

Reply:
See line 498. We are directly writing `frequency_bits` to sysfs but sysfs only accepts 0 or 1. In your case, you should write 0 to sysfs, but you will write 0b00.

Reply:
Yea, you are right. Fixed it.